src/dmd/expressionsem.d

```diff
@@ -7734,28 +7742,57 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
                         if (e2x.isLvalue())
                         {
-                            if (!e2x.type.implicitConvTo(e1x.type))
+                            if (sd.copyCtor)
                             {
-                                exp.error("conversion error from `%s` to `%s`",
-                                    e2x.type.toChars(), e1x.type.toChars());
-                                return setError();
+                                /* Rewrite as:
+                                 * e1 = init, e1.copyCtor(e2);
+                                 */
+                                Expression einit = new BlitExp(exp.loc, exp.e1, getInitExp(sd, exp.loc, sc, t1));
+                                einit.type = e1x.type;
+
+                                if (!einit)
+                                    return setError();
+
+                                Expression e;
```

Comment:
Insert a comment showing what expression is being built.

Reply:
Isn't the one on line 7742 enough?

Reply:
It's nearly 300 lines away, so no.